### PR TITLE
Make it easier for users to discover that copying arrays and structs to storage is unimplemented only in legacy codegen and works via IR

### LINKED ISSUE #12783: `UnimplementedFeatureError` with no line indication when copying an array of structs to storage
## Description

I am getting an error with no indication of which line it refers to:
```
UnimplementedFeatureError: Copying of type struct MyStruct memory[] memory to storage not yet supported.
```

## Environment

- Compiler version: 0.8.12

## Steps to Reproduce

Well, I need to dig through my code in order to find it, but on the compiler side, the problem is probably in ArrayUtils.cpp:
```
solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage not yet supported.");
```

Thanks :)

### PR DESCRIPTION
**Pull request by mehtavishwa30**
Closes #12783. The message is the only part that's not already covered by #15139.

### CODE REVIEW

File: libsolidity/codegen/ArrayUtils.cpp [hunk 1]
```diff
@@ -224,7 +224,7 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 				else if (sourceBaseType->isValueType())
 					CompilerUtils(_context).loadFromMemoryDynamic(*sourceBaseType, fromCalldata, true, false);
 				else
-					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage not yet supported.");
+					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage not yet supported in legacy codegen. Note: It is implemented in via IR.");
```
Comment (cameel) [inline libsolidity/codegen/ArrayUtils.cpp]: How about this?

```suggestion
					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage is only supported by the IR pipeline. Hint: try compiling with `--via-ir` (CLI) or the equivalent `viaIR: true` (Standard JSON)");
```

Comment (mehtavishwa30) [reply libsolidity/codegen/ArrayUtils.cpp]: Much better. Although, this is an error for unimplemented features. Shouldn't the primary focus of the error message be to describe where it's not implemented i.e. legacy? and then as a side note include that they can try it with via ir?

Comment (cameel) [reply libsolidity/codegen/ArrayUtils.cpp]: Well, in this case if it's not one then it's the other so not much difference IMO. Just thought it sounded a bit better this way. But if you prefer, I'm fine with your version for the first sentence too. TBH I mostly wanted to fix the weirdly sounding "in via IR" bit and also list the flags like we do in the StackTooDeep message :)

Comment (mehtavishwa30) [reply libsolidity/codegen/ArrayUtils.cpp]: How about now?

**Review by cameel — APPROVED** [review]

**Comment by Imrall** [comment]
| Header | Header | Header |
|--------|--------|--------|
| Cell | Cell | Cell |
| Cell | Cell | Cell |
| Cell | Cell | Cell |

**Pull request merged**